app/views/devise/mailer/confirmation_instructions.html.erb

```diff
@@ -1,5 +1,6 @@
-<p>Welcome <%= @email %>!</p>
+<% require 'devise/version' %>
+<p><%= t('.greeting', :recipient => @resource.email, :default => "Welcome #{@resource.email}!") %></p>
 
-<p>You can confirm your account email through the link below:</p>
-
-<p><%= link_to 'Confirm my account', confirmation_url(@resource, :confirmation_token => @token) %></p>
+<p><%= t('.instruction', :default => "You can confirm your account email through the link below:") %></p>
+<p><%= link_to t('.action', :default => "Confirm my account"),
+         confirmation_url(@resource, :confirmation_token => (Devise::VERSION.start_with?('3.') ? @token : @resource.confirmation_token)) %></p>
```

Comment:
I don't know how to answer your questions. This was auto-generated by ```devise-i18n-views```, which I suppose knows better than I what's the best in this case.